src/mongoc/mongoc-socket.h

```diff
@@ -38,6 +38,12 @@
 # include <sys/un.h>
 #endif
 
+#ifdef __hpux__
+typedef int       mongoc_socklen_t;
```

Comment:
we should probably move this to autoconf check, I can image more unixes not having `socklen_t`